### PR TITLE
Support getting configuration secrets from vault

### PR DESCRIPTION
**Pull request by cdoron**

### CODE REVIEW

File: abm/connector.py [hunk 1]
```diff
@@ -25,6 +26,16 @@ def __init__(self, config, logger, workdir):
...
 
+        if 'vault_credentials' in self.config:
+            vault_credentials = self.config['vault_credentials']
+            del self.config['vault_credentials']
+            secrets = get_secrets_from_vault(vault_credentials=vault_credentials, datasetID=asset_name)
+            if secrets:
+                # merge config with secrets returned by vault
+                self.config = dict(self.config, **secrets)
+            else:
+                logger.info("no secrets returned by vault")
```
Comment (revit13) [inline abm/connector.py:37]: should it be an error in case the creds are missing?

File: abm/vault.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#
+# Copyright 2023 IBM Corp.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from fybrik_python_vault import get_jwt_from_file, get_raw_secret_from_vault
+from fybrik_python_logging import logger, DataSetID, ForUser
```
Comment (revit13) [inline abm/vault.py:7]: its seems that the logger is not used in this file.

File: abm/connector.py [hunk 3]
```diff
@@ -25,6 +26,16 @@ def __init__(self, config, logger, workdir):
         if 'connector' not in self.config:
             raise ValueError("'connector' field missing from configuration")
 
+        if 'vault_credentials' in self.config:
+            vault_credentials = self.config['vault_credentials']
+            del self.config['vault_credentials']
+            secrets = get_secrets_from_vault(vault_credentials=vault_credentials, datasetID=asset_name)
+            if secrets:
```
Comment (revit13) [inline abm/connector.py:33]: maybe add a comment that not all connection types require creds. Later an error can be raised based on the connection type - for example for mysql where creds are required.

Comment (cdoron) [reply abm/connector.py:33]: In my opinion, the `airbyte-module` should not be aware of the different asset types. Ideally, it should not know which configuration fields are required for `mysql` (for example).

**Review by revit13 — APPROVED** [review]

**Pull request merged**